servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/RequestTagExtractor.java

```diff
@@ -32,15 +34,22 @@ private static String getRequestMethod(HttpRequestMetaData req) {
     }
 
     private static String getHttpUrl(HttpRequestMetaData req) {
-        return (req.scheme() == null ? "http" : req.scheme()) + "://" +
-            (req.effectiveHostAndPort() == null ? "localhost:8080" : req.effectiveHostAndPort())
-            + req.rawPath()
+        return req.path()
             + (req.rawQuery() == null ? "" : "?" + req.rawQuery());
     }
 
     static Span reportTagsAndStart(SpanBuilder span, HttpRequestMetaData httpRequestMetaData) {
         span.setAttribute("http.url", getHttpUrl(httpRequestMetaData));
         span.setAttribute("http.method", getRequestMethod(httpRequestMetaData));
+        span.setAttribute("http.target", getHttpUrl(httpRequestMetaData));
+        span.setAttribute("http.route", httpRequestMetaData.rawPath());
+        span.setAttribute("http.flavor", httpRequestMetaData.version().major() + "."
+            + httpRequestMetaData.version().minor());
+        Optional.ofNullable(httpRequestMetaData.userInfo())
```

Comment:
Do we need optionals here or can we just use a regular if not null?

Reply:
(consider avoiding Optional allocations on the hot code path if we don't have to do it)

Reply:
That makes sense. I removed them.